packages/iov-bns/src/bnsconnection.spec.ts

```diff
@@ -318,7 +326,7 @@ describe("BnsConnection", () => {
       expect(mine.primarySignature.nonce).toEqual(nonce);
       expect(mine.primarySignature.signature.length).toBeTruthy();
       expect(mine.otherSignatures.length).toEqual(0);
-      const tx = mine.transaction;
+      const tx = mine.transaction as BnsTx;
```

Comment:
I like this. Better yet, we have an `isBnsTx()` guard to check it matches on runtime, and maybe `requireBnsTx()` helper that matches the guard or throws an error.